x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/pytorch/process/PyTorchStateStreamer.java

```diff
@@ -70,6 +72,14 @@ public void writeStateToStream(String modelId, String index, OutputStream restor
         restorer.setSearchSize(1);
         restorer.restoreModelDefinition(doc -> writeChunk(doc, restoreStream), success -> {
             logger.debug("model [{}] state restored in [{}] documents from index [{}]", modelId, restorer.getNumDocsWritten(), index);
+            if (bytesWritten.get() != modelSize) {
+                logger.error(
+                    "model [{}] restored state size [{}] does not equal the expected model size [{}]",
+                    modelId,
+                    bytesWritten,
+                    modelSize
+                );
```

Comment:
I like this. It shows user if their estimate size is off! Good thinking.

But, I think if we validate this on _start deployment, that is enough: https://github.com/elastic/elasticsearch/pull/80439

Reply:
Actually, NVM. The validation in start deployment makes sure the total definition length is represented by the parts. 

If we move the model size estimation to be provided on configuration PUT, then it is wise to validate and log like this when the process starts up.